scripts/gen_image_generic.sh

```diff
@@ -47,12 +59,11 @@ dos_dircopy() {
 }
 
 [ -n "$PADDING" ] && dd if=/dev/zero of="$OUTPUT" bs=512 seek="$ROOTFSOFFSET" conv=notrunc count="$ROOTFSSIZE"
+[ -n "$GUID" ] && dd if="$OUTPUT.grub.img" of="$OUTPUT" bs=512 seek=512 conv=notrunc count=2048
 dd if="$ROOTFSIMAGE" of="$OUTPUT" bs=512 seek="$ROOTFSOFFSET" conv=notrunc
 
 if [ -n "$GUID" ]; then
-    [ -n "$PADDING" ] && dd if=/dev/zero of="$OUTPUT" bs=512 seek="$((ROOTFSOFFSET + ROOTFSSIZE))" conv=notrunc count="$sect"
-    mkfs.fat --invariant -n kernel -C "$OUTPUT.kernel" -S 512 "$((KERNELSIZE / 1024))"
-    LC_ALL=C dos_dircopy "$KERNELDIR" /
+    make_ext4fs -J -L openwrt_boot -l "$KERNELSIZE" ${SOURCE_DATE_EPOCH:+-T ${SOURCE_DATE_EPOCH}} "$OUTPUT.kernel" "$KERNELDIR"
```

Comment:
This doesn't look right.  You're trying to create /boot, separate from / and you want this to be ext4?  Seems this should be a separate commit.

Reply:
Yes, you might be right, I'm not even sure if ext4 was really needed.